docs/website/contents/for-developers/HardWonWisdom.md

```diff
@@ -210,31 +210,41 @@ The possibility of an never-ending era is only supported for tests and for reuse
-Thus, we cannot handle a translation query unless we know which era its argument---be it a slot, an epoch, or a UTC time---is in and the durations of all preceding eras, or, equivalently, the start times of all eras up-to-and-including the argument is in.
+Thus, we cannot handle a translation query unless we know which era its argument---be it a slot, an epoch, or a UTC time---is in and the durations of all preceding eras, or, equivalently, the start times of all eras up-to-and-including the one the argument is in.
 
 Each query is handled by some piece of code and with respect to some ledger state.
 The code (including config files) determines the supported eras (including their epoch size and slot length), and the ledger state determines which prefix of eras has known start times.
 The known start times are enough to recognize when the given argument is during some era with a known end time, in which case the translation succeeds.
 (The translation starts from the chain's start time, adds the duration of all preceding eras, and then adds the relative time since the beginning of the argument's era.)
 Otherwise, we need more information to know whether the given argument is during the last era with a known start time, ie the first era with an unknown end time.
-When the query argument is after that start time, the fact that the era's end is not yet known means that the query argument maybe either be in that era or instead be in a future era with a different epoch size/slot length, and so the correct translation may differ from the hypothetical response derived from the assumption the query argument precedes the first unknown end time.
+When the query argument is after that start time, the fact that the era's end is not yet known means that the query argument is either in that era or instead in a future era with a possibly different epoch size/slot length, and so the correct translation may differ from the hypothetical response derived from the assumption the query argument precedes the first unknown end time.
 
-For the first era with an unknown end time (ie the last era with a known start time), the code relies an a lower bound for that end time, ie the soonest that era could possibly end (aka "the horizon" or "the forecast horizon").
-If the query argument precedes that horizon, then the translation succeeds, otherwise it fails.
-For this purpose, the code (including config files) cannot support an era unless it also knows a "safe zone" for that era, where the safe zone (along with any additional assumptions the Consensus code makes) is enough information to determine the horizon induced by a given ledger state.
+For a ledger state in the first era with an unknown end time (ie the last era with a known start time), the code relies an a lower bound for that end time, ie the soonest that era could possibly end.
+If a query argument handled by that ledger state precedes that bound, then the translation succeeds, otherwise it fails.
```

Comment:
Are we referring to a query translation here?

Reply:
I think I figured it out, and I added a few more words to try to clarify.